testsuite/cluster-test/src/health/fullnode_check.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{
+    cluster::Cluster,
+    health::{HealthCheck, HealthCheckContext},
+    instance::Instance,
+};
+use async_trait::async_trait;
+use futures::future::join_all;
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+use std::env;
+
+pub static THRESHOLD: Lazy<i64> = Lazy::new(|| {
+    if let Ok(v) = env::var("FULL_NODE_HEALTH_THRESHOLD") {
+        v.parse()
+            .expect("Failed to parse FULL_NODE_HEALTH_THRESHOLD")
+    } else {
+        20000_i64
+    }
+});
+
+pub struct FullNodeHealthCheck {
+    cluster: Cluster,
+}
+
+impl FullNodeHealthCheck {
+    pub fn new(cluster: Cluster) -> FullNodeHealthCheck {
+        Self { cluster }
+    }
+}
+
+async fn get_version(instance: &Instance) -> (&Instance, i64) {
+    let res = instance
+        .debug_interface_client()
+        .get_node_metric("libra_state_sync_committed_version{}")
+        .await;
+    let content = match res {
+        Ok(res) => res.unwrap_or_default(),
+        _ => 0i64,
+    };
+    (&instance, content)
+}
+
+#[async_trait]
+impl HealthCheck for FullNodeHealthCheck {
+    async fn verify(&mut self, ctx: &mut HealthCheckContext) {
+        let validator = self.cluster.validator_instances();
+        let fullnode = self.cluster.fullnode_instances();
+
+        let futures = validator.iter().map(get_version);
+        let val_latest_version = join_all(futures).await;
+        let val_latest_version: HashMap<String, i64> = val_latest_version
+            .into_iter()
+            .map(|(instance, version)| (instance.validator_index(), version))
+            .collect();
+
+        let futures = fullnode.iter().map(get_version);
+        let fullnode_latest_version = join_all(futures).await;
+
+        for (fullnode, fullnode_version) in fullnode_latest_version {
+            let index = fullnode.validator_index();
+            let val_version = val_latest_version[&index];
```

Comment:
Btw, I think you want to use some method like `get` here, I think `[]` might just panic if key is not found?
We want to avoid panics in code like this